### PR TITLE
Update `create` doc on allowed `vals_list` types

### PR DESCRIPTION
**Pull request by AdrianTeng**
Though `create` actually accepts taking a `dict`, and this is stated in the details description in the docstring, because the param type is currently only `list`, when you do pass a `dict` into `create`, your IDE (such as PyCharm) might warn you on this:

image

With this change IDE would understands both types are allow and stops complaining.

image

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

**Comment by MiquelRForgeFlow** [comment]
@rco-odoo

File: odoo/models.py [hunk 1]
```diff
@@ -3970,7 +3970,7 @@ def create(self, vals_list):
         The new records are initialized using the values from the list of dicts
         ``vals_list``, and if necessary those from :meth:`~.default_get`.
 
-        :param list vals_list:
+        :param Union[list, dict] vals_list:
```
Comment (rco-odoo) [inline odoo/models.py:3973]: The right type is actually a **`list` of `dict`** or a `dict`.

Comment (AdrianTeng) [reply odoo/models.py]: @rco-odoo updated!

**Review by rco-odoo — APPROVED** [review]
@robodoo rebase-ff r+

**Comment by rco-odoo** [comment]
@robodoo retry

**Pull request closed without merging**